train.py

```diff
@@ -70,28 +70,40 @@ def print_progress_bar(iteration, total, time_per_step, prefix='', suffix='', le
         loss.backward()
  
         optimizer.step()
- 
-        total_loss += loss.item()  # Accumulate loss value
- 
+  
     # Loop through steps with a progress bar
     for step in range(1, total_steps + 1):
         # Simulate some processing time (replace this with your actual training code)
         time.sleep(time_per_step)
  
         # Display the progress bar
         print_progress_bar(step, total_steps, time_per_step, prefix=f'Epoch {epoch}/{total_epochs}', length=30, fill='=')
-   
-    average_loss = total_loss / len(train_dataset)  # Calculate average loss for the epoch
-    print(f'\nAverage Loss: {average_loss:.4f}')
- 
-    # Add a new line after completing an epoch
-    print()
- 
+
+    val_loss = 0.0
+    for data in val_dataset:
+        landmarks = torch.tensor([element for row in data['landmarks'] for element in row], dtype=torch.float)
+        ans = [0 for _ in range(TARGET_SIZE)]
+        ans[data['gesture']] = 1
+        ans = torch.tensor(ans, dtype=torch.float)
+        res = model(landmarks)
+        loss = loss_function(res, ans)
+        val_loss += loss.item()  # Accumulate loss value
+
+    average_loss = val_loss / len(val_dataset)  # Calculate average loss for the epoch
+    print(f'\nValidation Loss: {average_loss:.4f}\n')
+
+    if prev_loss < average_loss:
+        tolerance -= 1
+        if tolerance == 0:
+            break
+

```

Comment:
Why don't we train the model to the end without middle pause?

Reply:
to avoid overfitting

Reply:
okay